### ISSUE TITLE
Report name of broken jar file

### ISSUE THREAD

**Issue by Artur-** [eclipse/jetty.project#783]
If you have corrupt jar file in your project, Jetty will fail to start with e.g.
```
java.util.zip.ZipException: invalid distance too far back
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
	at java.util.zip.ZipInputStream.read(ZipInputStream.java:194)
	at java.util.jar.JarInputStream.read(JarInputStream.java:207)
	at java.util.zip.ZipInputStream.closeEntry(ZipInputStream.java:140)
	at java.util.zip.ZipInputStream.getNextEntry(ZipInputStream.java:118)
	at java.util.jar.JarInputStream.getNextEntry(JarInputStream.java:142)
	at java.util.jar.JarInputStream.getNextJarEntry(JarInputStream.java:179)
	at org.eclipse.jetty.annotations.AnnotationParser.parseJar(AnnotationParser.java:939)
	at org.eclipse.jetty.annotations.AnnotationParser.parse(AnnotationParser.java:851)
```

If this exception message contained also the invalid JAR file, it would help a lot when trying to hunt it down.

Tested with 9.3.11.v20160721

**Comment by janbartel** [comment]
If you enable debug logging for org.eclipse.jetty.annotations package you'll see that we print out which jar file is being parsed.

**Comment by Artur-** [comment]
Right, if you realize that you can/should do that. On the other hand if
```entry = jar_in.getNextJarEntry();```
was wrapped in the same or a similar try catch as "parseJarEntry", the exception could make much more sense without needing to tune debug logging

**Comment by janbartel** [comment]
@Artur-  yep, you're correct, I can't argue with that :)

**Comment by janbartel** [comment]
Fixed in jetty-9.3 and beyond.
